src/JuliusSweetland.OptiKey.Core/GazeSmoothingFilter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) 2020 OPTIKEY LTD (UK company number 11854839) - All Rights Reserved
+using JuliusSweetland.OptiKey.Models;
+using JuliusSweetland.OptiKey.Properties;
+using System;
+using System.Windows;
+
+namespace JuliusSweetland.OptiKey.DataFilters
+{
+    public class GazeSmoothingFilter
+
+    {
+        private double EstimationNoise;
+        readonly double ProcessNoise;
+        private double Gain;
+        private double SmoothingLevel;
+        private Point EstimatedValue;
+
+        public GazeSmoothingFilter()
+        {
+            EstimatedValue = new Point(0, 0);
+            EstimationNoise = 1000;
+            ProcessNoise = 500;
+        }
+
+        public Point Update(Point measuredValue)
+        {
+            SmoothingLevel = Settings.Default.GazeSmoothingLevel;
+            var smoothingBase = (measuredValue - EstimatedValue).Length / SmoothingLevel;
```

Comment:
smoothingBase = distanceMoved = delta
This is the base number we will use in our exponential

Reply:
@AdamRoden If you coud decorate the code with these comments that would be very helpful.